### PR TITLE
feat(needle):add needle to scale

### LINKED ISSUE #4917: The issue of using scale to replace meter
### Introduce the problem
In v9, the **meter**  has been removed and replaced with **scale.** However, if you want to use **needle,** currently this is achieved by adding a needle image and setting an angle. In V8, the meter can directly set numerical values, and the needle can display the corresponding position, making it very convenient to use.
### Suggested solution
Therefore, should we restore the **meter** control or add a **needle** control? This way, users can also reduce the difficulty of use.

### PR DESCRIPTION
**Pull request by lhdjply**
### Description of the feature or fix

#4917
feat(needle):add needle to scale

image

### Checkpoints
- [ ] Run `code-format.py` from the scripts folder. [astyle](http://astyle.sourceforge.net/install.html) needs to be installed.
- [ ] Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed
- [ ] Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant. 
- [ ] Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- [ ] If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/release/v8.3/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/release/v8.3/Kconfig).
 
Be sure the following conventions are followed:
- [ ] Follow the [Styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [ ] Prefer `enum`s instead of macros. If inevitable to use `define`s export them with `LV_EXPORT_CONST_INT(defined_value)` right after the `define`.
- [ ] In function arguments prefer `type name[]` declaration for array parameters instead of `type * name`
- [ ] Use typed pointers instead of `void *` pointers
- [ ] Do not `malloc` into a static or global variables. Instead declare the variable in `lv_global_t` structure in [`lv_global.h`](https://github.com/lvgl/lvgl/blob/master/src/core/lv_global.h) and mark the variable with `(LV_GLOBAL_DEFAULT()->variable)` when it's used. See a detailed description [here](https://docs.lvgl.io/master/get-started/bindings/micropython.html#memory-management).
- [ ] Widget constructor must follow the `lv_<widget_name>_create(lv_obj_t * parent)` pattern.
- [ ] Widget members function must start with `lv_<module_name>` and should receive `lv_obj_t *` as first argument which is a pointer to widget object itself.  
- [ ] `struct`s should be used via an API and not modified directly via their elements.
- [ ] `struct` APIs should follow the widgets' conventions. That is to receive a pointer to the `struct` as the first argument, and the prefix of the `struct` name should be used as the prefix of the function name too (e.g.  `lv_disp_set_default(lv_disp_t * disp)`)
- [ ] Functions and `struct`s which are not part of the public API must begin with underscore in order to mark them as "private".
- [ ] Arguments must be named in H files too.
- [ ] To register and use callbacks one of the following needs to be followed (see a detailed description [here](https://docs.lvgl.io/master/get-started/bindings/micropython.html#callbacks)):
  - For both the registration function and the callback pass a pointer to a `struct` as the first argument. The `struct` must contain `void * user_data` field.
  - The last argument of the registration function must be `void * user_data` and the same `user_data` needs to be passed as the last argument of the callback.
  - Callback types not following these conventions should end with `xcb_t`.

### CODE REVIEW

**Comment by lhdjply** [comment]
1. Currently, using images to create needs has not been implemented yet.
2. Is there a way to add a circle to the starting point of the line by setting the style?

**Comment by kisvegabor** [comment]
Thank you for the PR.

I think it'd be more flexible to approach it like this:
```c
lv_obj_t * scale = lv_scale_create(parent);

lv_obj_t * line = lv_line_create(scale);
//Add styles to the line

lv_scale_set_line_needle_value(scale, line, value, needle_length);
```

This way we could add multiple needles to the scale as well.

**Comment by lhdjply** [comment]
make sense

File: src/widgets/scale/lv_scale.c [hunk 1]
```diff
@@ -182,6 +182,47 @@ void lv_scale_set_round_props(lv_obj_t * obj, uint32_t angle_range, int32_t rota
     lv_obj_invalidate(obj);
 }
 
+void lv_scale_set_line_needle_value(lv_obj_t * obj, lv_obj_t * needle_line, uint32_t needle_length, int32_t value)
+{
+    uint32_t angle;
+    lv_value_precise_t scale_width, scale_height;
+    lv_value_precise_t needle_length_x, needle_length_y;
+    float cosx_angle_value, sinx_angle_value;
```
Comment (kisvegabor) [inline src/widgets/scale/lv_scale.c]: `float` shouldn't be used. Here a simple `int32_t` should be enough too, as we need only 1 px precision.

File: src/widgets/scale/lv_scale.c [hunk 2]
```diff
@@ -182,6 +182,47 @@ void lv_scale_set_round_props(lv_obj_t * obj, uint32_t angle_range, int32_t rota
...
+    else {
+        angle = scale->angle_range * (value - scale->range_min) / (scale->range_max - scale->range_min);
+    }
+
+    scale_width = (lv_value_precise_t)(lv_obj_get_style_width(obj, LV_PART_MAIN));
+    scale_height = (lv_value_precise_t)(lv_obj_get_style_height(obj, LV_PART_MAIN));
+
+    cosx_angle_value = (float)(lv_trigo_cos(scale->rotation + angle)) / 32768;
+    sinx_angle_value = (float)(lv_trigo_sin(scale->rotation + angle)) / 32768;
+    needle_length_x = (lv_value_precise_t)(needle_length * cosx_angle_value);
```
Comment (kisvegabor) [inline src/widgets/scale/lv_scale.c]: This should wourk without `float`
```c 
needle_length_x = (needle_length * lv_trigo_cos(scale->rotation + angle))  >> LV_TRIGO_SHIFT;
```

**Comment by lhdjply** [comment]
Operation error, changed the hash value of someone else's commit. I closed this PR and resubmitted a new one

**Pull request closed without merging**